api/v1alpha1/packagebundle.go

```diff
@@ -26,22 +26,58 @@ func (config *PackageBundle) ExpectedKind() string {
 	return PackageBundleKind
 }
 
-func (config *PackageBundle) FindSource(pkgName, pkgVersion string) (retSource PackageOCISource, err error) {
+func (config *PackageBundle) FindPackage(pkgName string) (retPkg BundlePackage, err error) {
 	for _, pkg := range config.Spec.Packages {
 		if strings.EqualFold(pkg.Name, pkgName) {
-			source := pkg.Source
-			for _, packageVersion := range source.Versions {
-				// We do not sort before getting `latest` because there will be only a single packageVersion per release in normal cases. For edge cases which may require multiple
-				// versions, the order in the file will be ordered according to what we want `latest` to point to
-				if packageVersion.Name == pkgVersion || packageVersion.Digest == pkgVersion || pkgVersion == Latest {
-					retSource = PackageOCISource{Registry: source.Registry, Repository: source.Repository, Digest: packageVersion.Digest, Version: packageVersion.Name}
-					return retSource, nil
-				}
-			}
+			return pkg, nil
+		}
+	}
+	return retPkg, fmt.Errorf("package not found in bundle (%s): %s", config.Name, pkgName)
+}
+
+func (config *PackageBundle) GetDependencies(version SourceVersion) (dependencies []BundlePackage, err error) {
+	var ret []BundlePackage
+	for _, dep := range version.Dependencies {
+		pkg, err := config.FindPackage(dep)
+		if err != nil {
+			return nil, err
+		}
+		ret = append(ret, pkg)
+	}
+	return ret, nil
+}
+
+func (config *PackageBundle) FindVersion(pkg BundlePackage, pkgVersion string) (ret SourceVersion, err error) {
+	source := pkg.Source
+	for _, packageVersion := range source.Versions {
+		// We do not sort before getting `latest` because there will be only a single packageVersion per release in normal cases. For edge cases which may require multiple
+		// versions, the order in the file will be ordered according to what we want `latest` to point to
+		if packageVersion.Name == pkgVersion || packageVersion.Digest == pkgVersion || pkgVersion == Latest {
+			return packageVersion, nil
 		}
 	}
+	return ret, fmt.Errorf("package version not found in bundle (%s): %s @ %s", config.Name, pkg.Name, pkgVersion)
+}
+
```

Comment:
Kind of unrelated to this PR, but I was feeling that OCISource was not enough information for the state machine.